modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/DataGenerationApplication.java

```diff
@@ -34,13 +35,13 @@ public DataGenerationApplication(Ignite ignite) {
     }
 
     /** {@inheritDoc} */
-    @Override protected void run(String[] args) {
+    @Override protected void run(JsonNode jsonNode) {
         log.info("Creating cache...");
 
-        IgniteCache<Integer, Integer> cache = ignite.createCache(args[0]);
+        IgniteCache<Integer, Integer> cache = ignite.createCache(jsonNode.get("cacheName").asText());
 
         try (IgniteDataStreamer<Integer, Integer> stmr = ignite.dataStreamer(cache.getName())) {
-            for (int i = 0; i < Integer.parseInt(args[1]); i++) {
+            for (int i = 0; i < jsonNode.get("range").asInt(); i++) {
```

Comment:
Is it possible to use a DTO object instead? Generics is more clear solution

Reply:
It will be unique for every app, this will lead to useless classes?

Reply:
Additional classes are ok. In case of complex hierarchy I'd prefer use generics (for example smth like `obj.getList().get(0).getRange()`) instead of ugly constructions (smth like `node.get("list").asArray()[0].get("range").asInt()`). 

DTOs make code much much clearer, also it provides sanity checks out of the box.

Reply:
Not sure I've got an idea.
Could you provide an example related to the current tests?

Reply:
Oh, I see. I thought that JsonNode is a generic, but it's not. So, there is a choose between plain JsonNode and DTO. In favor of latter:
1.  Currently there are not, but in one day could be complex json with nested objects and arrays. User will have to use `mapper.readValue(jsonNode, DTO.class)` if need it. 
2. If you want change code of your app in one day and change params, there won't help from IDE for you. There is a possibility of NPE in jsonNode.get(). DTO provides checks for such situations (code fails in compile time).
3. All tests have own package, so I don't see problems with additional classes that are isolated within a package.

Reply:
How to share java DTO to python? Seems we should add and sync DTO couple for each application. 
How to sync this automatically?
Seem overcomplicated...?

Reply:
I think, that we should make in the bright future a generic application, that accepts broad range of generic options to load data to cache. For example, cache name, key-value schema (binary object), range for keys, number of keys, etc.

For this argument DTO is a preferrable solution, I suppose. But we can implement this a little bit later as a separate task.

Reply:
There still python, shame :( There are different approaches. Python just use dict and user is responsible for everything that is running in python code. Java could provide more strict checks. 

Ok, agree, not an issue, as python won't directly use DTO object.

Reply:
Great, let's re-discuss this later